hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -1256,11 +1257,12 @@ public HFileBlock readBlock(long dataBlockOffset, long onDiskBlockSize, final bo
 
     BlockCacheKey cacheKey =
       new BlockCacheKey(name, dataBlockOffset, this.isPrimaryReplicaReader(), expectedBlockType);
+    Attributes attributes = Attributes.of(BLOCK_CACHE_KEY_KEY, cacheKey.toString());
 
     boolean useLock = false;
     IdLock.Entry lockEntry = null;
-    Span span = TraceUtil.getGlobalTracer().spanBuilder("HFileReaderImpl.readBlock").startSpan();
-    try (Scope traceScope = span.makeCurrent()) {
+    final Span span = Span.current();
```

Comment:
@Apache9 I believe we have discussed the potential performance impacts of open spans for every block. This patch does away with this approach, instead, we use span events to mark the results of block-level operations on the span currently in context. Please let me know if you have any concerns with this approach.

Reply:
Let's just do a benchmark?

Reply:
For the curious, a benchmark was performed and posted over on https://github.com/apache/hbase/pull/4601#issuecomment-1195598276